harness/determined/exec/harness.py

```diff
@@ -224,4 +226,8 @@ def main() -> None:
 
 
 if __name__ == "__main__":
-    main()
+    try:
```

Comment:
blocking: you should actually catch the exception on running `prepare_controller()` and `controller.run()` not at the top level.

Reply:
Actually, I think this is the right choice.  When we catch an exception during the `trial.__init__()`, we need to send a response to the master and just exit.  We could call "sys.exit()", but we already catch sys.exit calls and assume the user messed up.  So instead the choice was to let the InvalidHP exception continue to bubble up.

But then you don't want to show an InvalidHP stacktrace necessarily, so we drop that exception here.

Reply:
Although it works, it looks complicated and hacky. Do you know how would this be simplified in a push-arch world?

Reply:
I think I changed this actually, in a recent commit so if you take another look at the harness.py file I don't think this is an issue anymore.